### ISSUE TITLE
Adding `--disable-kube-proxy` does not seem to get added to service file for k3s

### ISSUE THREAD

**Issue by sdwilsh** [kairos-io/kairos#1962]
**Kairos version:**

```
PRETTY_NAME="Ubuntu 22.04.3 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.3 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy
KAIROS_NAME="kairos-kairos-ubuntu-22-lts"
KAIROS_VERSION="v2.4.1-k3sv1.27.3+k3s1"
KAIROS_ID="kairos"
KAIROS_ID_LIKE="kairos-kairos-ubuntu-22-lts"
KAIROS_VERSION_ID="v2.4.1-k3sv1.27.3+k3s1"
KAIROS_PRETTY_NAME="kairos-kairos-ubuntu-22-lts v2.4.1-k3sv1.27.3+k3s1"
KAIROS_BUG_REPORT_URL="https://github.com/kairos-io/kairos/issues"
KAIROS_HOME_URL="https://github.com/kairos-io/kairos"
KAIROS_IMAGE_REPO="quay.io/kairos/kairos-ubuntu-22-lts"
KAIROS_IMAGE_LABEL=""
KAIROS_GITHUB_REPO="kairos-io/provider-kairos"
KAIROS_VARIANT="kairos"
KAIROS_FLAVOR="ubuntu-22-lts"
```

**CPU architecture, OS, and Version:**

```
Linux node01.hogs.tswn.us 6.2.0-33-generic #33~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Thu Sep  7 10:33:52 UTC 2 x86_64 x86_64 x86_64 GNU/Linux
```

**Describe the bug**

Adding `--disable-kube-proxy` to `k3s.args` in `/oem/90_custom.yaml` is not reflected in `/etc/systemd/system/k3s.service.d/override.conf`.

**To Reproduce**

1. Add `- --disable-kube-proxy` to `/oem/90_custom.yaml` under `k3s.args`.
2. Reboot
3. See in logs for k3s that kube-proxy did run

**Expected behavior**

kube-proxy should be disabled, and the service should pass `--disable-kube-proxy`.

**Logs**

**Additional context**

I spun this out of https://github.com/kairos-io/kairos/discussions/1958 since @jimmykarily implied there might be an issue here with kairos-provider.

I've stripped out the `--etcd-s3` args I have since there's a secret-key there, but here's what `k3s.args` is:
```
k3s:
  args:
    - --node-ip=10.11.0.101
    - --node-name=node01
    - --cluster-init
    - --flannel-backend=none
    - --disable-network-policy
    - --cluster-cidr=172.16.0.0/16
    - --service-cidr=172.17.0.0/16
    - --disable=traefik,servicelb
    - --disable-kube-proxy
    - --tls-san=10.11.0.100
    - --tls-san=k3s.hogs.tswn.us
 ```
Here's the `/etc/systemd/system/k3s.service.d/override.conf` file, again, I stripped out the `--etcd-s3` arguments:
```
[Service]
ExecStart=
ExecStart=/usr/bin/k3s server --node-ip=10.11.0.101 --node-name=node01 --cluster-init --flannel-backend=none --disable-network-policy --cluster-cidr=172.16.0.0/16 --service-cidr=172.17.0.0/16 --disable=traefik,servicelb --tls-san=10.11.0.100 --tls-san=k3s.hogs.tswn.us
```

This is the only control plane in the cluster of three total machines right now.

**Comment by sdwilsh** [comment]
I think this is expected based on the code.  The service file is only updated during one-time bootstrapping: https://github.com/kairos-io/provider-kairos/blob/ee4547deb9e207cdb8a336fc1e8f3dac3991be91/internal/provider/bootstrap.go#L224

If that is intentional, it's not clear in the docs: https://kairos.io/docs/reference/configuration/

**Comment by jimmykarily** [comment]
To clarify, if the flag was there from the first installation, it would work right?
Given the `/etc/systemd` directory is mounted on COS_PERSISTENT, you can edit it directly and add the missing flag manually and it will persist reboots.

We could make the `k3s` block be evaluated on every boot and update services and such, but I'm not sure all possible configuration can be added just like that. Maybe some won't be respected unless other steps that occur during first bootstrap happen.

**Comment by sdwilsh** [comment]
Yeah, it's just not well documented that this block is only relevant for first boot, and otherwise one needs to modify the service file themselves.  I spent longer than I'd like to admit trying to figure out why my changes were not applying, before I finally realized this is never updated.

I'd be fine with a documentation update to consider this issue resolved if that's the path we want to take.  Bundles might have the same issue.
